### PR TITLE
Update App to work with latest branch updates

### PR DESCRIPTION
**Pull request by jfrerich**
The main purpose of this PR is to update the app to work with the latest updates for the following repos/branches.
* `mm-plugin-apps` `master`
* `mm-redux` `feature/cloud-apps`
* `mm-webapp` `feature/cloud-apps`

Other changes include:
* remove `call_handlers` folder
  * this was used when using `type` to determine the call handling
  * now each `restapi/fxxx` contains the direct calls needed per routing path

* standardize call path routes  
  * route variable names and routes

- [x] TODO: Current store implementation only store JSON data. Need to support `json` and `string` values.  Config is `json`, but an auth `token` does not need to be `json`
* ticket created.  https://mattermost.atlassian.net/browse/MM-34285

Also resolves the following tickets:
- [x] https://mattermost.atlassian.net/browse/MM-33541 - add ephemeral message after saving / updating zendesk config
- [x] https://mattermost.atlassian.net/browse/MM-33386 - used forked version of `node-zendesk`
  * submitted PR to author
- [x] https://mattermost.atlassian.net/browse/MM-34099 - Cannot add bot to team that is not open for anyone to join
- [x] https://mattermost.atlassian.net/browse/MM-34105 - The word "ok" appears in red text below the Type filed on submission attempts
- [x] https://mattermost.atlassian.net/browse/MM-34106 - HTML error shows when selecting and executing create ticket command 
- [x] https://mattermost.atlassian.net/browse/MM-34102 - Modal will not close after ticket creation
- [x] https://mattermost.atlassian.net/browse/MM-34100 - Ticket links are constructed wrong in the feedback post made by the bot

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Comment by jfrerich** [comment]
@mickmister and @catalintomai, would y'all mind giving this PR a lookover?  This PR syncs with the latest branches for mm-redux, mm-webapp (feature/cloud-apps branches), and plugin-cloud-apps master.  Once this is merged, I can start submitting smaller PRs that are focussed on bugs :)

File: src/restapi/fConfig.ts [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+    const callResponse = newFormCallResponse(form);
+    res.json(callResponse);
+}
+
+export async function fSubmitOrUpdateZendeskConfigSubmit(req: Request, res: Response): Promise<void> {
+    const call: AppCall = req.body;
+
+    let callResponse: AppCallResponse = newOKCallResponseWithMarkdown('Successfully updated Zendesk configuration');
+    try {
+        const context = call.context;
```
Comment (catalintomai) [inline src/restapi/fConfig.ts:26]: nit: is there a need for `context` ?

File: src/restapi/fSubscriptions.ts [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+// OpenSubscriptionsForm opens a new subscriptions form
 export async function fOpenSubscriptionsForm(req: Request, res: Response): Promise<void> {
-    const callResponse = await new OpenSubscriptionsForm(req.body).handle();
+    const form = await newSubscriptionsForm(req.body);
+    const callResponse = newFormCallResponse(form);
     res.json(callResponse);
 }
 
+// SubmitOrUpdateSubscriptionsForm updates the subscriptions form with new values or
+// submits the the form if submit button is clicked
```
Comment (catalintomai) [inline src/restapi/fSubscriptions.ts:15]: s/'the the'/'the'

File: src/restapi/fConfig.ts (hunk 1 above)
Comment (jfrerich) [reply src/restapi/fConfig.ts]: no.  removed.

File: src/restapi/fSubscriptions.ts (hunk 2 above)
Comment (jfrerich) [reply src/restapi/fSubscriptions.ts]: done.

File: src/restapi/mm_routes.ts [hunk 3]
```diff
@@ -19,21 +19,27 @@ const router = express.Router();
...
-router.post(Routes.App.BindingPathOpenSubscriptionsForm, fOpenSubscriptionsForm);
-router.post(Routes.App.BindingPathOpenZendeskConfigForm, fOpenZendeskConfigForm);
-router.post(Routes.App.BindingPathHelp, fHelp);
-
-// Callable routes, not bound to a location
-router.post(Routes.App.CallPathSubmitOrUpdateCreateTicketForm, fSubmitOrUpdateCreateTicketForm);
-router.post(Routes.App.CallPathSubmitOrUpdateSubscriptionForm, fSubmitOrUpdateSubscriptionsForm);
-router.post(Routes.App.CallPathSubmitOrUpdateZendeskConfigForm, fSubmitOrUpdateZendeskConfigForm);
+// formless calls
+router.post(Routes.App.BindingPathConnectSubmit, fConnect);
```
Comment (catalintomai) [inline src/restapi/mm_routes.ts:25]: 1/5 s/`BindingPathConnectSubmit `/`connect/submit` (we don't use the const anywhere and no need to search for it - inline-ing it should be ok) + the others.

**Review by catalintomai — DISMISSED** [review]

File: src/restapi/mm_routes.ts (hunk 3 above)
Comment (jfrerich) [reply src/restapi/mm_routes.ts]: Removed all `/submit` and `/form` endpoints from `constants.ts` file. These are routes that only need to be defined one time. The routes that remain in
`utils/constants.ts` `Routes.App` are call paths that can be used by multiple locations

**Pull request merged**